asset/js/setup.js

```diff
@@ -32,6 +32,34 @@ function setupWithSearch(siteData) {
   VueStrap.installEvents(vm);
 }
 
+/**
+ *   site-nav-btn function to toggle itself and site navigation elements.
+ */
+// eslint-disable-next-line no-unused-vars
+function siteNavToggle(siteNavBtn) {
+  siteNavBtn.classList.toggle('morph');
+  document.getElementById('site-nav').classList.toggle('open');
+  document.getElementById('site-nav-btn-wrap').classList.toggle('open');
+}
+
 jQuery.getJSON(`${baseUrl}/siteData.json`)
   .then(siteData => setupWithSearch(siteData))
   .catch(() => setup());
+
+/**
+ * Creates event listener for all dropdown-btns in page.
+ */
+Array.prototype.forEach.call(document.getElementsByClassName('dropdown-btn'),
+                             element => element.addEventListener('click', function () {
+                               this.classList.toggle('dropdown-open');
+                               const dropdownContent = this.nextElementSibling;
+                               if (dropdownContent.style.maxHeight === '0px'
+                                 || !dropdownContent.style.maxHeight) {
+                                 dropdownContent.style.transition = 'max-height 0.25s ease-in';
+                                 dropdownContent.style.maxHeight = '1000px';
+                               } else {
+                                 dropdownContent.style.transition = 'max-height 0.15s ease-out';
+                                 dropdownContent.style.maxHeight = '0px';
+                               }
+                             }));
+
```

Comment:
- When dealing with huge blocks of code with lots of whitespace, prefer:

  ```js
  Array.prototype.forEach.call(
    document.getElementsByClassName('dropdown-btn'),
    element => element.addEventListener('click', () => {
      this.classList.toggle('dropdown-open');
      const dropdownContent = this.nextElementSibling;
      if (dropdownContent.style.maxHeight === '0px'
        || !dropdownContent.style.maxHeight) {
        dropdownContent.style.transition = 'max-height 0.25s ease-in';
        dropdownContent.style.maxHeight = '1000px';
      } else {
        dropdownContent.style.transition = 'max-height 0.15s ease-out';
        dropdownContent.style.maxHeight = '0px';
      }
    }));
  ```

- Extra empty line at end of file.